### PR TITLE
[ZEPPELIN-3526] Zeppelin auth mechanisms (LDAP or password based) should be mutually exclusive

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
### What is this PR for?
Problem:
When any external authentication (like LDAP/AD) is enabled for Zeppelin, the default password-based authentication could still be configured in addition to that. This makes space for backdoor in Zeppelin where the user can still get in using the local username/password.

Proposed Solution:
Zeppelin shouldn't allow specifying [users] section in shiro.ini when it is configured to authenticate with LDAP/AD.

### What type of PR is it?
[Bug Fix | Feature ]

### Todos
* [x] - Add documentation 

### What is the Jira issue?
* [ZEPPELIN-3526](https://issues.apache.org/jira/browse/ZEPPELIN-3526)

### How should this be tested?
If both [users] and [main] for example activeDirectoryRealm section enabled in shiro, Zeppelin server should not start.

### CODE REVIEW

**Comment by prabhjyotsingh** [comment]
@Leemoonsoo @felixcheung @zjffdu  can you help review this.

**Comment by zjffdu** [comment]
LGTM, one suggestion is that I think it is time for us to do refactoring for the authentication component, we should put it in one component and provide interface to be used by other components. Instead of putting logic in zeppelin server like this PR.

**Comment by prabhjyotsingh** [comment]
Agreed we should definitely do something like zeppelin-plugins for authentication components as well in Zeppelin-0.9

**Comment by mebelousov** [comment]
@prabhjyotsingh I think it's good to document only one way for authentification. What do you think about it?

File: docs/setup/security/shiro_authentication.md [hunk 1]
```diff
@@ -267,13 +268,15 @@ If you want to grant this permission to other users, you can change **roles[ ]**
 
 ### Apply multiple roles in Shiro configuration
 By default, Shiro will allow access to a URL if only user is part of "**all the roles**" defined like this:
+
```
Comment (prabhjyotsingh) [inline docs/setup/security/shiro_authentication.md]: Added empty lines for http://zeppelin.apache.org/docs/0.8.0-SNAPSHOT/setup/security/shiro_authentication.html#apply-multiple-roles-in-shiro-configuration

**Comment by prabhjyotsingh** [comment]
@mebelousov sure I've added a doc.

**Pull request closed without merging**